File: 3264-maximum-points-after-enemy-battles/3264-maximum-points-after-enemy-battles.py
class Solution:
    def maximumPoints(self, enemyEnergies: List[int], currentEnergy: int) -> int:
        mini = min(enemyEnergies)
        if(mini>currentEnergy):
            return 0
        sum1 = currentEnergy
        for i in enemyEnergies:
            sum1+=i
        return((sum1-mini)//mini)

File: 3264-maximum-points-after-enemy-battles/3264-maximum-points-after-enemy-battles.cpp
class Solution {
public:
    long long maximumPoints(vector<int>& enemyEnergies, int currentEnergy) {
        sort(enemyEnergies.begin(),enemyEnergies.end());
        int n = enemyEnergies.size();
        long low = 0;
        long high = n-1;
        long long count = 0;
        while(low<=high){
            if(currentEnergy>=enemyEnergies[low]){
                count+= currentEnergy/enemyEnergies[low];
                currentEnergy = currentEnergy%enemyEnergies[low];
                
            }
            else if(count>0){
                currentEnergy = currentEnergy + enemyEnergies[high];
                high--;
            }
            else{
                break;
            }
        }
        return count;
        
    }
};